applications/tari_base_node/src/builder.rs

```diff
@@ -474,19 +472,19 @@ where
         .set_base_node_public_key(base_node_public_key.clone())
         .await
         .expect("Problem setting local base node public key for transaction service.");
-    transaction_service_handle
-        .restart_transaction_protocols()
-        .await
-        .expect("Problem restarting transaction protocols in the Transaction Service");
-    // Only start the transaction broadcast protocols once the local node is synced
+    let oms_handle = wallet_handles
+        .get_handle::<OutputManagerHandle>()
+        .expect("OutputManagerService is not registered");
+    // Only start the transaction broadcast and UTXO validartion protocols once the local node is synced
     let state_machine = base_node_handles
         .get_handle::<StateMachineHandle>()
         .expect("Could not get State Machine handle");
+    let config_base_node_query_timeout_clone = config.base_node_query_timeout.clone();
     task::spawn(async move {
```

Comment:
So instead of spawning an `async{ ... }` block here put all this code into an async function and then you can just call spawn on that function.

Reply:
Agreed